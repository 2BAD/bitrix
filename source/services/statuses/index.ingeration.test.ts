/* tslint-env jest */
// tslint:disable: no-expression-statement object-literal-sort-keys

import Bitrix from './../../bitrix'

const isSortedAsc = (arr: ReadonlyArray<string>) => arr.slice(1).every((v, i) => arr[i] <= v)
const isSortedDesc = (arr: ReadonlyArray<string>) => arr.slice(1).every((v, i) => arr[i] >= v)

const WEBHOOK_URL = process.env.WEBHOOK_URL

// tslint:disable-next-line: no-if-statement
if (!WEBHOOK_URL) {
  // tslint:disable-next-line: no-throw
  throw Error('Integration tests require environmental variable `WEBHOOK_URL` to be set')
}

const { statuses } = Bitrix(WEBHOOK_URL)

describe('Statuses', () => {
  describe('fields', () => {
    it('should get all fields', async () => {
      const { result } = await statuses.fields()
      expect(result).toMatchSnapshot()
    })
  })

  describe('list', () => {
    it('should get all entries without optional parameters', async () => {
      const { result } = await statuses.list()
      expect(result).toMatchSnapshot()
    })

    it('should return only entries with ENTITY_ID = STATUS sorted in ascending order', async () => {
      const { result } = await statuses.list({
        order: { SORT: 'ASC' },
        filter: { ENTITY_ID: 'STATUS' }
      })
      expect(result.every((s) => s.ENTITY_ID === 'STATUS')).toBeTruthy()
      expect(isSortedAsc(result.map((s) => s.SORT))).toBeTruthy()
      expect(result).toMatchSnapshot()
    })

    it('should return only entries with ENTITY_ID = STATUS sorted in descending order', async () => {
      const { result } = await statuses.list({
        order: { SORT: 'DESC' },
        filter: { ENTITY_ID: 'STATUS' }
      })
      expect(result.every((s) => s.ENTITY_ID === 'STATUS')).toBeTruthy()
      expect(isSortedDesc(result.map((s) => s.SORT))).toBeTruthy()
      expect(result).toMatchSnapshot()
    })
  })

  describe('get', () => {
    it('should get entry with ID = 1', async () => {
      const { result } = await statuses.get('1')
      expect(result).toMatchSnapshot()
    })
    it('should return an error in case non existent ID is requested', async () => {
      const get = statuses.get('1001')
      await expect(get).rejects.toThrow(Error)
    })
  })

  describe('create', () => {
    it('should create an entry', async () => {
      const entry = {
        ENTITY_ID: 'STATUS',
        STATUS_ID: 'TEST',
        NAME: 'TEST'
      }

      const { result: id } = await statuses.create(entry)
      expect(id).toBeGreaterThan(0)

      const { result: status } = await statuses.get(id)
      expect(status).toMatchSnapshot({
        ID: expect.any(String)
      })

      const { result: deleteResult } = await statuses.delete(id)
      expect(deleteResult).toBeTruthy()
    })
  })

  describe('update', () => {
    it('should update an entry', async () => {
      const STATUS_ID = 101
      const { result: status } = await statuses.get(STATUS_ID)
      expect(status).toMatchSnapshot()

      const { result: updateResult } = await statuses.update(STATUS_ID, { ...status, NAME: 'CHANGED' })
      expect(updateResult).toBeTruthy()

      const { result: updatedStatus } = await statuses.get(STATUS_ID)
      expect(updatedStatus).toMatchSnapshot()

      const { result: rollbackResult } = await statuses.update(STATUS_ID, status)
      expect(rollbackResult).toBeTruthy()
    })
  })
})
