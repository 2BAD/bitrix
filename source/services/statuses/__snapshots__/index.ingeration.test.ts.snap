// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Statuses create should create an entry 1`] = `
Object {
  "ENTITY_ID": "STATUS",
  "EXTRA": Object {
    "SEMANTICS": "process",
  },
  "ID": Any<String>,
  "NAME": "TEST",
  "NAME_INIT": "",
  "SORT": "10",
  "STATUS_ID": "TEST",
  "SYSTEM": "N",
}
`;

exports[`Statuses get should get entry with ID = 1 1`] = `
Object {
  "ENTITY_ID": "STATUS",
  "EXTRA": Object {
    "COLOR": "#39A8EF",
    "SEMANTICS": "process",
  },
  "ID": "1",
  "NAME": "Не обработан",
  "NAME_INIT": "Не обработан",
  "SORT": "10",
  "STATUS_ID": "NEW",
  "SYSTEM": "Y",
}
`;

exports[`Statuses list should get all entries without optional parameters 1`] = `
Array [
  Object {
    "ENTITY_ID": "DEAL_STAGE",
    "EXTRA": Object {
      "COLOR": "#39A8EF",
      "SEMANTICS": "process",
    },
    "ID": "101",
    "NAME": "Новая",
    "NAME_INIT": "Новая",
    "SORT": "10",
    "STATUS_ID": "NEW",
    "SYSTEM": "Y",
  },
  Object {
    "ENTITY_ID": "EVENT_TYPE",
    "ID": "125",
    "NAME": "Информация",
    "NAME_INIT": "Информация",
    "SORT": "10",
    "STATUS_ID": "INFO",
    "SYSTEM": "Y",
  },
  Object {
    "ENTITY_ID": "EMPLOYEES",
    "ID": "53",
    "NAME": "менее 50",
    "NAME_INIT": "менее 50",
    "SORT": "10",
    "STATUS_ID": "EMPLOYEES_1",
    "SYSTEM": "Y",
  },
  Object {
    "ENTITY_ID": "CONTACT_TYPE",
    "ID": "35",
    "NAME": "Клиенты",
    "NAME_INIT": "",
    "SORT": "10",
    "STATUS_ID": "CLIENT",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "DEAL_STATE",
    "ID": "117",
    "NAME": "В планах",
    "NAME_INIT": "",
    "SORT": "10",
    "STATUS_ID": "PLANNED",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "COMPANY_TYPE",
    "ID": "43",
    "NAME": "Клиент",
    "NAME_INIT": "",
    "SORT": "10",
    "STATUS_ID": "CUSTOMER",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "CALL_LIST",
    "ID": "61",
    "NAME": "В работе",
    "NAME_INIT": "В работе",
    "SORT": "10",
    "STATUS_ID": "IN_WORK",
    "SYSTEM": "Y",
  },
  Object {
    "ENTITY_ID": "DEAL_TYPE",
    "ID": "91",
    "NAME": "Продажа",
    "NAME_INIT": "Продажа",
    "SORT": "10",
    "STATUS_ID": "SALE",
    "SYSTEM": "Y",
  },
  Object {
    "ENTITY_ID": "INDUSTRY",
    "ID": "69",
    "NAME": "Информационные технологии",
    "NAME_INIT": "",
    "SORT": "10",
    "STATUS_ID": "IT",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "QUOTE_STATUS",
    "EXTRA": Object {
      "COLOR": "#39A8EF",
      "SEMANTICS": "process",
    },
    "ID": "131",
    "NAME": "Новое",
    "NAME_INIT": "Новое",
    "SORT": "10",
    "STATUS_ID": "DRAFT",
    "SYSTEM": "Y",
  },
  Object {
    "ENTITY_ID": "SOURCE",
    "ID": "11",
    "NAME": "Звонок",
    "NAME_INIT": "Звонок",
    "SORT": "10",
    "STATUS_ID": "CALL",
    "SYSTEM": "Y",
  },
  Object {
    "ENTITY_ID": "STATUS",
    "EXTRA": Object {
      "COLOR": "#39A8EF",
      "SEMANTICS": "process",
    },
    "ID": "1",
    "NAME": "Не обработан",
    "NAME_INIT": "Не обработан",
    "SORT": "10",
    "STATUS_ID": "NEW",
    "SYSTEM": "Y",
  },
  Object {
    "ENTITY_ID": "HONORIFIC",
    "ID": "141",
    "NAME": "г-н",
    "NAME_INIT": "",
    "SORT": "10",
    "STATUS_ID": "HNR_RU_1",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "EMPLOYEES",
    "ID": "55",
    "NAME": "50-250",
    "NAME_INIT": "",
    "SORT": "20",
    "STATUS_ID": "EMPLOYEES_2",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "QUOTE_STATUS",
    "EXTRA": Object {
      "COLOR": "#2FC6F6",
      "SEMANTICS": "process",
    },
    "ID": "133",
    "NAME": "Отправлено клиенту",
    "NAME_INIT": "",
    "SORT": "20",
    "STATUS_ID": "SENT",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "STATUS",
    "EXTRA": Object {
      "COLOR": "#55D0E0",
      "SEMANTICS": "process",
    },
    "ID": "3",
    "NAME": "В работе",
    "NAME_INIT": "",
    "SORT": "20",
    "STATUS_ID": "IN_PROCESS",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "COMPANY_TYPE",
    "ID": "45",
    "NAME": "Поставщик",
    "NAME_INIT": "",
    "SORT": "20",
    "STATUS_ID": "SUPPLIER",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "HONORIFIC",
    "ID": "143",
    "NAME": "г-жа",
    "NAME_INIT": "",
    "SORT": "20",
    "STATUS_ID": "HNR_RU_2",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "INDUSTRY",
    "ID": "71",
    "NAME": "Телекоммуникации и связь",
    "NAME_INIT": "",
    "SORT": "20",
    "STATUS_ID": "TELECOM",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "DEAL_STATE",
    "ID": "119",
    "NAME": "В работе",
    "NAME_INIT": "В работе",
    "SORT": "20",
    "STATUS_ID": "PROCESS",
    "SYSTEM": "Y",
  },
  Object {
    "ENTITY_ID": "CONTACT_TYPE",
    "ID": "37",
    "NAME": "Поставщики",
    "NAME_INIT": "",
    "SORT": "20",
    "STATUS_ID": "SUPPLIER",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "DEAL_TYPE",
    "ID": "93",
    "NAME": "Комплексная продажа",
    "NAME_INIT": "",
    "SORT": "20",
    "STATUS_ID": "COMPLEX",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "EVENT_TYPE",
    "ID": "127",
    "NAME": "Телефонный звонок",
    "NAME_INIT": "Телефонный звонок",
    "SORT": "20",
    "STATUS_ID": "PHONE",
    "SYSTEM": "Y",
  },
  Object {
    "ENTITY_ID": "DEAL_STAGE",
    "EXTRA": Object {
      "COLOR": "#2FC6F6",
      "SEMANTICS": "process",
    },
    "ID": "103",
    "NAME": "Подготовка документов",
    "NAME_INIT": "",
    "SORT": "20",
    "STATUS_ID": "PREPARATION",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "CALL_LIST",
    "ID": "63",
    "NAME": "Успешно",
    "NAME_INIT": "Успешно",
    "SORT": "20",
    "STATUS_ID": "SUCCESS",
    "SYSTEM": "Y",
  },
  Object {
    "ENTITY_ID": "SOURCE",
    "ID": "13",
    "NAME": "Электронная почта",
    "NAME_INIT": "",
    "SORT": "20",
    "STATUS_ID": "EMAIL",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "DEAL_STAGE",
    "EXTRA": Object {
      "COLOR": "#55D0E0",
      "SEMANTICS": "process",
    },
    "ID": "105",
    "NAME": "Cчёт на предоплату",
    "NAME_INIT": "",
    "SORT": "30",
    "STATUS_ID": "PREPAYMENT_INVOICE",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "CALL_LIST",
    "ID": "65",
    "NAME": "Неверный номер",
    "NAME_INIT": "Неверный номер",
    "SORT": "30",
    "STATUS_ID": "WRONG_NUMBER",
    "SYSTEM": "Y",
  },
  Object {
    "ENTITY_ID": "STATUS",
    "EXTRA": Object {
      "COLOR": "#47E4C2",
      "SEMANTICS": "process",
    },
    "ID": "5",
    "NAME": "Обработан",
    "NAME_INIT": "",
    "SORT": "30",
    "STATUS_ID": "PROCESSED",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "QUOTE_STATUS",
    "EXTRA": Object {
      "COLOR": "#7BD500",
      "SEMANTICS": "success",
    },
    "ID": "135",
    "NAME": "Принято",
    "NAME_INIT": "Принято",
    "SORT": "30",
    "STATUS_ID": "APPROVED",
    "SYSTEM": "Y",
  },
  Object {
    "ENTITY_ID": "DEAL_TYPE",
    "ID": "95",
    "NAME": "Продажа товара",
    "NAME_INIT": "",
    "SORT": "30",
    "STATUS_ID": "GOODS",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "EVENT_TYPE",
    "ID": "129",
    "NAME": "Отправлен email",
    "NAME_INIT": "Отправлен email",
    "SORT": "30",
    "STATUS_ID": "MESSAGE",
    "SYSTEM": "Y",
  },
  Object {
    "ENTITY_ID": "SOURCE",
    "ID": "15",
    "NAME": "Веб-сайт",
    "NAME_INIT": "",
    "SORT": "30",
    "STATUS_ID": "WEB",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "CONTACT_TYPE",
    "ID": "39",
    "NAME": "Партнеры",
    "NAME_INIT": "",
    "SORT": "30",
    "STATUS_ID": "PARTNER",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "COMPANY_TYPE",
    "ID": "47",
    "NAME": "Конкурент",
    "NAME_INIT": "",
    "SORT": "30",
    "STATUS_ID": "COMPETITOR",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "DEAL_STATE",
    "ID": "121",
    "NAME": "Выполнена",
    "NAME_INIT": "Выполнена",
    "SORT": "30",
    "STATUS_ID": "COMPLETE",
    "SYSTEM": "Y",
  },
  Object {
    "ENTITY_ID": "EMPLOYEES",
    "ID": "57",
    "NAME": "250-500",
    "NAME_INIT": "",
    "SORT": "30",
    "STATUS_ID": "EMPLOYEES_3",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "INDUSTRY",
    "ID": "73",
    "NAME": "Производство",
    "NAME_INIT": "",
    "SORT": "30",
    "STATUS_ID": "MANUFACTURING",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "DEAL_STAGE",
    "EXTRA": Object {
      "COLOR": "#47E4C2",
      "SEMANTICS": "process",
    },
    "ID": "107",
    "NAME": "В работе",
    "NAME_INIT": "",
    "SORT": "40",
    "STATUS_ID": "EXECUTING",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "DEAL_STATE",
    "ID": "123",
    "NAME": "Отменена",
    "NAME_INIT": "Отменена",
    "SORT": "40",
    "STATUS_ID": "CANCELED",
    "SYSTEM": "Y",
  },
  Object {
    "ENTITY_ID": "QUOTE_STATUS",
    "EXTRA": Object {
      "COLOR": "#FF5752",
      "SEMANTICS": "failure",
    },
    "ID": "137",
    "NAME": "Отклонено",
    "NAME_INIT": "Отклонено",
    "SORT": "40",
    "STATUS_ID": "DECLAINED",
    "SYSTEM": "Y",
  },
  Object {
    "ENTITY_ID": "DEAL_TYPE",
    "ID": "97",
    "NAME": "Продажа услуги",
    "NAME_INIT": "",
    "SORT": "40",
    "STATUS_ID": "SERVICES",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "SOURCE",
    "ID": "17",
    "NAME": "Реклама",
    "NAME_INIT": "",
    "SORT": "40",
    "STATUS_ID": "ADVERTISING",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "INDUSTRY",
    "ID": "75",
    "NAME": "Банковские услуги",
    "NAME_INIT": "",
    "SORT": "40",
    "STATUS_ID": "BANKING",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "EMPLOYEES",
    "ID": "59",
    "NAME": "более 500",
    "NAME_INIT": "",
    "SORT": "40",
    "STATUS_ID": "EMPLOYEES_4",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "CALL_LIST",
    "ID": "67",
    "NAME": "Больше не звонить",
    "NAME_INIT": "Больше не звонить",
    "SORT": "40",
    "STATUS_ID": "STOP_CALLING",
    "SYSTEM": "Y",
  },
  Object {
    "ENTITY_ID": "COMPANY_TYPE",
    "ID": "49",
    "NAME": "Партнер",
    "NAME_INIT": "",
    "SORT": "40",
    "STATUS_ID": "PARTNER",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "CONTACT_TYPE",
    "ID": "41",
    "NAME": "Другое",
    "NAME_INIT": "",
    "SORT": "40",
    "STATUS_ID": "OTHER",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "STATUS",
    "EXTRA": Object {
      "COLOR": "#7BD500",
      "SEMANTICS": "success",
    },
    "ID": "7",
    "NAME": "Качественный лид",
    "NAME_INIT": "Качественный лид",
    "SORT": "40",
    "STATUS_ID": "CONVERTED",
    "SYSTEM": "Y",
  },
  Object {
    "ENTITY_ID": "DEAL_STAGE",
    "EXTRA": Object {
      "COLOR": "#FFA900",
      "SEMANTICS": "process",
    },
    "ID": "109",
    "NAME": "Финальный счёт",
    "NAME_INIT": "",
    "SORT": "50",
    "STATUS_ID": "FINAL_INVOICE",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "COMPANY_TYPE",
    "ID": "51",
    "NAME": "Другое",
    "NAME_INIT": "",
    "SORT": "50",
    "STATUS_ID": "OTHER",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "QUOTE_STATUS",
    "EXTRA": Object {
      "COLOR": "#FF5752",
      "SEMANTICS": "apology",
    },
    "ID": "139",
    "NAME": "Анализ причины отклонения",
    "NAME_INIT": "",
    "SORT": "50",
    "STATUS_ID": "APOLOGY",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "SOURCE",
    "ID": "19",
    "NAME": "Существующий клиент",
    "NAME_INIT": "",
    "SORT": "50",
    "STATUS_ID": "PARTNER",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "DEAL_TYPE",
    "ID": "99",
    "NAME": "Сервисное обслуживание",
    "NAME_INIT": "",
    "SORT": "50",
    "STATUS_ID": "SERVICE",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "INDUSTRY",
    "ID": "77",
    "NAME": "Консалтинг",
    "NAME_INIT": "",
    "SORT": "50",
    "STATUS_ID": "CONSULTING",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "STATUS",
    "EXTRA": Object {
      "COLOR": "#FF5752",
      "SEMANTICS": "failure",
    },
    "ID": "9",
    "NAME": "Некачественный лид",
    "NAME_INIT": "Некачественный лид",
    "SORT": "50",
    "STATUS_ID": "JUNK",
    "SYSTEM": "Y",
  },
  Object {
    "ENTITY_ID": "SOURCE",
    "ID": "21",
    "NAME": "По рекомендации",
    "NAME_INIT": "",
    "SORT": "60",
    "STATUS_ID": "RECOMMENDATION",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "DEAL_STAGE",
    "EXTRA": Object {
      "COLOR": "#7BD500",
      "SEMANTICS": "success",
    },
    "ID": "111",
    "NAME": "Сделка успешна",
    "NAME_INIT": "Сделка успешна",
    "SORT": "60",
    "STATUS_ID": "WON",
    "SYSTEM": "Y",
  },
  Object {
    "ENTITY_ID": "INDUSTRY",
    "ID": "79",
    "NAME": "Финансы",
    "NAME_INIT": "",
    "SORT": "60",
    "STATUS_ID": "FINANCE",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "INDUSTRY",
    "ID": "81",
    "NAME": "Правительство",
    "NAME_INIT": "",
    "SORT": "70",
    "STATUS_ID": "GOVERNMENT",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "DEAL_STAGE",
    "EXTRA": Object {
      "COLOR": "#FF5752",
      "SEMANTICS": "failure",
    },
    "ID": "113",
    "NAME": "Сделка провалена",
    "NAME_INIT": "Сделка провалена",
    "SORT": "70",
    "STATUS_ID": "LOSE",
    "SYSTEM": "Y",
  },
  Object {
    "ENTITY_ID": "SOURCE",
    "ID": "23",
    "NAME": "Выставка",
    "NAME_INIT": "",
    "SORT": "70",
    "STATUS_ID": "TRADE_SHOW",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "SOURCE",
    "ID": "25",
    "NAME": "CRM-форма",
    "NAME_INIT": "CRM-форма",
    "SORT": "75",
    "STATUS_ID": "WEBFORM",
    "SYSTEM": "Y",
  },
  Object {
    "ENTITY_ID": "SOURCE",
    "ID": "27",
    "NAME": "Обратный звонок",
    "NAME_INIT": "Обратный звонок",
    "SORT": "77",
    "STATUS_ID": "CALLBACK",
    "SYSTEM": "Y",
  },
  Object {
    "ENTITY_ID": "SOURCE",
    "ID": "29",
    "NAME": "Генератор продаж",
    "NAME_INIT": "Генератор продаж",
    "SORT": "78",
    "STATUS_ID": "RC_GENERATOR",
    "SYSTEM": "Y",
  },
  Object {
    "ENTITY_ID": "SOURCE",
    "ID": "31",
    "NAME": "Интернет-магазин",
    "NAME_INIT": "Интернет-магазин",
    "SORT": "79",
    "STATUS_ID": "STORE",
    "SYSTEM": "Y",
  },
  Object {
    "ENTITY_ID": "DEAL_STAGE",
    "EXTRA": Object {
      "COLOR": "#FF5752",
      "SEMANTICS": "apology",
    },
    "ID": "115",
    "NAME": "Анализ причины провала",
    "NAME_INIT": "",
    "SORT": "80",
    "STATUS_ID": "APOLOGY",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "SOURCE",
    "ID": "33",
    "NAME": "Другое",
    "NAME_INIT": "",
    "SORT": "80",
    "STATUS_ID": "OTHER",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "INDUSTRY",
    "ID": "83",
    "NAME": "Доставка",
    "NAME_INIT": "",
    "SORT": "80",
    "STATUS_ID": "DELIVERY",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "INDUSTRY",
    "ID": "85",
    "NAME": "Развлечения",
    "NAME_INIT": "",
    "SORT": "90",
    "STATUS_ID": "ENTERTAINMENT",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "INDUSTRY",
    "ID": "87",
    "NAME": "Не для получения прибыли",
    "NAME_INIT": "",
    "SORT": "100",
    "STATUS_ID": "NOTPROFIT",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "INVOICE_STATUS",
    "ID": "144",
    "NAME": "Новый",
    "NAME_INIT": "Новый",
    "SORT": "100",
    "STATUS_ID": "N",
    "SYSTEM": "Y",
  },
  Object {
    "ENTITY_ID": "INDUSTRY",
    "ID": "89",
    "NAME": "Другое",
    "NAME_INIT": "Другое",
    "SORT": "110",
    "STATUS_ID": "OTHER",
    "SYSTEM": "Y",
  },
  Object {
    "ENTITY_ID": "INVOICE_STATUS",
    "ID": "150",
    "NAME": "Отправлен клиенту",
    "NAME_INIT": "",
    "SORT": "110",
    "STATUS_ID": "S",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "INVOICE_STATUS",
    "ID": "148",
    "NAME": "Оплачен",
    "NAME_INIT": "Оплачен",
    "SORT": "120",
    "STATUS_ID": "P",
    "SYSTEM": "Y",
  },
  Object {
    "ENTITY_ID": "INVOICE_STATUS",
    "ID": "146",
    "NAME": "Не оплачен",
    "NAME_INIT": "Не оплачен",
    "SORT": "130",
    "STATUS_ID": "D",
    "SYSTEM": "Y",
  },
]
`;

exports[`Statuses list should return only entries with ENTITY_ID = STATUS sorted in ascending order 1`] = `
Array [
  Object {
    "ENTITY_ID": "STATUS",
    "EXTRA": Object {
      "COLOR": "#39A8EF",
      "SEMANTICS": "process",
    },
    "ID": "1",
    "NAME": "Не обработан",
    "NAME_INIT": "Не обработан",
    "SORT": "10",
    "STATUS_ID": "NEW",
    "SYSTEM": "Y",
  },
  Object {
    "ENTITY_ID": "STATUS",
    "EXTRA": Object {
      "COLOR": "#55D0E0",
      "SEMANTICS": "process",
    },
    "ID": "3",
    "NAME": "В работе",
    "NAME_INIT": "",
    "SORT": "20",
    "STATUS_ID": "IN_PROCESS",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "STATUS",
    "EXTRA": Object {
      "COLOR": "#47E4C2",
      "SEMANTICS": "process",
    },
    "ID": "5",
    "NAME": "Обработан",
    "NAME_INIT": "",
    "SORT": "30",
    "STATUS_ID": "PROCESSED",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "STATUS",
    "EXTRA": Object {
      "COLOR": "#7BD500",
      "SEMANTICS": "success",
    },
    "ID": "7",
    "NAME": "Качественный лид",
    "NAME_INIT": "Качественный лид",
    "SORT": "40",
    "STATUS_ID": "CONVERTED",
    "SYSTEM": "Y",
  },
  Object {
    "ENTITY_ID": "STATUS",
    "EXTRA": Object {
      "COLOR": "#FF5752",
      "SEMANTICS": "failure",
    },
    "ID": "9",
    "NAME": "Некачественный лид",
    "NAME_INIT": "Некачественный лид",
    "SORT": "50",
    "STATUS_ID": "JUNK",
    "SYSTEM": "Y",
  },
]
`;

exports[`Statuses list should return only entries with ENTITY_ID = STATUS sorted in descending order 1`] = `
Array [
  Object {
    "ENTITY_ID": "STATUS",
    "EXTRA": Object {
      "COLOR": "#FF5752",
      "SEMANTICS": "failure",
    },
    "ID": "9",
    "NAME": "Некачественный лид",
    "NAME_INIT": "Некачественный лид",
    "SORT": "50",
    "STATUS_ID": "JUNK",
    "SYSTEM": "Y",
  },
  Object {
    "ENTITY_ID": "STATUS",
    "EXTRA": Object {
      "COLOR": "#7BD500",
      "SEMANTICS": "success",
    },
    "ID": "7",
    "NAME": "Качественный лид",
    "NAME_INIT": "Качественный лид",
    "SORT": "40",
    "STATUS_ID": "CONVERTED",
    "SYSTEM": "Y",
  },
  Object {
    "ENTITY_ID": "STATUS",
    "EXTRA": Object {
      "COLOR": "#47E4C2",
      "SEMANTICS": "process",
    },
    "ID": "5",
    "NAME": "Обработан",
    "NAME_INIT": "",
    "SORT": "30",
    "STATUS_ID": "PROCESSED",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "STATUS",
    "EXTRA": Object {
      "COLOR": "#55D0E0",
      "SEMANTICS": "process",
    },
    "ID": "3",
    "NAME": "В работе",
    "NAME_INIT": "",
    "SORT": "20",
    "STATUS_ID": "IN_PROCESS",
    "SYSTEM": "N",
  },
  Object {
    "ENTITY_ID": "STATUS",
    "EXTRA": Object {
      "COLOR": "#39A8EF",
      "SEMANTICS": "process",
    },
    "ID": "1",
    "NAME": "Не обработан",
    "NAME_INIT": "Не обработан",
    "SORT": "10",
    "STATUS_ID": "NEW",
    "SYSTEM": "Y",
  },
]
`;

exports[`Statuses update should update an entry 1`] = `
Object {
  "ENTITY_ID": "DEAL_STAGE",
  "EXTRA": Object {
    "COLOR": "#39A8EF",
    "SEMANTICS": "process",
  },
  "ID": "101",
  "NAME": "Новая",
  "NAME_INIT": "Новая",
  "SORT": "10",
  "STATUS_ID": "NEW",
  "SYSTEM": "Y",
}
`;

exports[`Statuses update should update an entry 2`] = `
Object {
  "ENTITY_ID": "DEAL_STAGE",
  "EXTRA": Object {
    "COLOR": "#39A8EF",
    "SEMANTICS": "process",
  },
  "ID": "101",
  "NAME": "CHANGED",
  "NAME_INIT": "Новая",
  "SORT": "10",
  "STATUS_ID": "NEW",
  "SYSTEM": "Y",
}
`;
